cmd/metal-api/internal/metal/switch.go

```diff
@@ -144,3 +162,203 @@ func (s *Switch) SetVrfOfMachine(m *Machine, vrf string) {
 	}
 	s.Nics = nics
 }
+
+// TranslateNicMap creates a NicMap where the keys are translated to the naming convention of the target OS
+//
+// example mapping from cumulus to sonic for one single port:
+//
+//	map[string]Nic {
+//		"swp1s1": Nic{
+//			Name: "Ethernet1",
+//			MacAddress: ""
+//		}
+//	}
+func (s *Switch) TranslateNicMap(targetOS SwitchOSVendor) (NicMap, error) {
+	nicMap := s.Nics.ByName()
+	translatedNicMap := make(NicMap)
+
+	if s.OS.Vendor == targetOS {
+		return nicMap, nil
+	}
+
+	ports := make([]string, 0)
+	for name := range nicMap {
+		ports = append(ports, name)
+	}
+
+	lines, err := getLinesFromPortNames(ports, s.OS.Vendor)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range ports {
+		targetPort, err := mapPortName(p, s.OS.Vendor, targetOS, lines)
+		if err != nil {
+			return nil, err
+		}
+
+		nic, ok := nicMap[p]
+		if !ok {
+			return nil, fmt.Errorf("an unknown error occured during port name translation")
+		}
+		translatedNicMap[targetPort] = nic
+	}
+
+	return translatedNicMap, nil
+}
+
+// MapPortNames creates a dictionary that maps the naming convention of this switch's OS to that of the target OS
+func (s *Switch) MapPortNames(targetOS SwitchOSVendor) (map[string]string, error) {
+	nics := s.Nics.ByName()
+	portNamesMap := make(map[string]string, len(s.Nics))
+
+	ports := make([]string, 0)
+	for name := range nics {
+		ports = append(ports, name)
+	}
+
+	lines, err := getLinesFromPortNames(ports, s.OS.Vendor)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range ports {
+		targetPort, err := mapPortName(p, s.OS.Vendor, targetOS, lines)
+		if err != nil {
+			return nil, err
+		}
+		portNamesMap[p] = targetPort
+	}
+
+	return portNamesMap, nil
+}
+
+func mapPortName(port string, sourceOS, targetOS SwitchOSVendor, allLines []int) (string, error) {
+	line, err := portNameToLine(port, sourceOS)
+	if err != nil {
+		return "", fmt.Errorf("unable to get line number from port name, %w", err)
+	}
+
+	if targetOS == SwitchOSVendorCumulus {
+		return cumulusPortByLineNumber(line, allLines), nil
+	}
+	if targetOS == SwitchOSVendorSonic {
+		return sonicPortByLineNumber(line), nil
+	}
+
+	return "", fmt.Errorf("unknown target switch os %s", targetOS)
+}
+
+func getLinesFromPortNames(ports []string, os SwitchOSVendor) ([]int, error) {
+	lines := make([]int, 0)
+	for _, p := range ports {
+		l, err := portNameToLine(p, os)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get line number from port name, %w", err)
+		}
+
+		lines = append(lines, l)
+	}
+
+	return lines, nil
+}
+
+func portNameToLine(port string, os SwitchOSVendor) (int, error) {
+	if os == SwitchOSVendorSonic {
+		return sonicPortNameToLine(port)
+	}
+	if os == SwitchOSVendorCumulus {
+		return cumulusPortNameToLine(port)
+	}
+	return 0, fmt.Errorf("unknown switch os %s", os)
```

Comment:
Prefer switch statement here to easier guarantee exhaustiveness as mentioned above.